source/tests/test_file_generator/test_file_generator.cpp

```diff
@@ -0,0 +1,196 @@
+#include "test_file_generator.h"
+
+test_file_generator::test_file_generator()
+{
+    sodium_init();
+    make_file_name();
+    set_random();
+    add_claim(char_file_name);
+    read_claim(char_file_name);
+    make_buffer();
+    add_stack(char_file_name);
+};
+
+test_file_generator::test_file_generator(string test_case, uint64_t stack_number)
+{
```

Comment:
I think it would be better to call delegated default constructor here.
May be something like this: 

```c++
test_file_generator::test_file_generator(string test_case, uint64_t stack_number) {
    test_file_generator()
    // here all different logic might be applied.
}
```